anitya/lib/backends/maven.py

```diff
@@ -70,3 +75,96 @@ def get_versions(cls, project):
                           .replace(':', '/'))
 
         return get_versions_by_regex(url, REGEX, project)
+
+    @classmethod
+    def verify_url(cls, group_id, artifact_id):
+        match_decimal = '(?=\d)'
+        match_non_decimal = r'(?=\D)'
+
+        # match name like like artifact_id23.23_1
+        temp_artifact = do_replace(artifact_id, match_decimal, match_decimal)
```

Comment:
Can you explain exactly what you want this regex to do? I _think_ you can rewrite this entire block with:
```
temp_artifact = artifact_id
temp_group = group_id
for pattern1, pattern2 in itertools.product(['(?=\d)', r'(?=\D)'], repeat=2):
    temp_artifact = re.sub(r'{}\.{}'.format(pattern1, pattern2), r'/', temp_artifact)
    temp_group = re.sub(r'{}\.{}'.format(pattern1, pattern2), r'/', temp_group)
```
But at first glance this seems to (eventually) match any ``.`` character since it'll match ``([0-9] | [a-zA-Z])\.([0-9] | [a-zA-Z])`` right? I'm generally a big fan of prefacing any regex with a comment about what I think it should be doing, since it's so easy to write confusing regex.